### PR TITLE
Sunset `wasmer-registry`, `wasmer-interface` and update misc dependencies

### LINKED ISSUE #5077: Lots of CVEs
### Describe the bug

There are lots of CVEs in the lock file.  Some of these might be dev-deps because osv-scanner isnt that smart, however some are in real deps.

#5076 for using a smarter tool to detect CVEs only in dependencies.

### Steps to reproduce

````
> osv-scanner --lockfile Cargo.lock
Scanned /home/jayvdb/rust/wasmer/Cargo.lock file and found 737 packages
╭─────────────────────────────────────┬──────┬───────────┬──────────────────┬─────────┬────────────╮
│ OSV URL                             │ CVSS │ ECOSYSTEM │ PACKAGE          │ VERSION │ SOURCE     │
├─────────────────────────────────────┼──────┼───────────┼──────────────────┼─────────┼────────────┤
│ https://osv.dev/GHSA-qc84-gqf4-9926 │ 8.1  │ crates.io │ crossbeam-utils  │ 0.7.2   │ Cargo.lock │
│ https://osv.dev/RUSTSEC-2022-0041   │      │           │                  │         │            │
│ https://osv.dev/GHSA-5wg8-7c9q-794v │ 5.5  │ crates.io │ lock_api         │ 0.3.4   │ Cargo.lock │
│ https://osv.dev/GHSA-gmv4-vmx3-x9f3 │      │           │                  │         │            │
│ https://osv.dev/GHSA-hj9h-wrgg-hgmx │      │           │                  │         │            │
│ https://osv.dev/GHSA-ppj3-7jw3-8vc4 │      │           │                  │         │            │
│ https://osv.dev/GHSA-vh4p-6j7g-f4j9 │      │           │                  │         │            │
│ https://osv.dev/RUSTSEC-2020-0070   │      │           │                  │         │            │
│ https://osv.dev/GHSA-wfg4-322g-9vqv │      │ crates.io │ memoffset        │ 0.5.6   │ Cargo.lock │
│ https://osv.dev/RUSTSEC-2023-0045   │      │           │                  │         │            │
│ https://osv.dev/RUSTSEC-2020-0016   │      │ crates.io │ net2             │ 0.2.39  │ Cargo.lock │
│ https://osv.dev/RUSTSEC-2024-0370   │      │ crates.io │ proc-macro-error │ 1.0.4   │ Cargo.lock │
│ https://osv.dev/GHSA-vr26-jcq5-fjj8 │ 8.7  │ crates.io │ quinn-proto      │ 0.11.3  │ Cargo.lock │
│ https://osv.dev/RUSTSEC-2024-0373   │      │           │                  │         │            │
│ https://osv.dev/GHSA-255r-3prx-mf99 │      │ crates.io │ rmp-serde        │ 0.15.5  │ Cargo.lock │
│ https://osv.dev/RUSTSEC-2022-0092   │      │           │                  │         │            │
│ https://osv.dev/RUSTSEC-2021-0127   │      │ crates.io │ serde_cbor       │ 0.11.2  │ Cargo.lock │
│ https://osv.dev/GHSA-wcg3-cvx6-7396 │ 6.2  │ crates.io │ time             │ 0.1.45  │ Cargo.lock │
│ https://osv.dev/RUSTSEC-2020-0071   │      │           │                  │         │            │
│ https://osv.dev/GHSA-fg7r-2g4j-5cgr │ 8.1  │ crates.io │ tokio            │ 0.1.22  │ Cargo.lock │
│ https://osv.dev/RUSTSEC-2021-0124   │      │           │                  │         │            │
│ https://osv.dev/RUSTSEC-2024-0320   │      │ crates.io │ yaml-rust        │ 0.4.5   │ Cargo.lock │
╰─────────────────────────────────────┴──────┴───────────┴──────────────────┴─────────┴────────────╯
```

### Expected behavior
<!-- A clear and concise description of what you expected to happen. -->

### Actual behavior

<!--
A clear and concise description of what actually happened.

If applicable, add screenshots to help explain your problem.
-->

### Additional context
<!-- Add any other context about the problem here. -->

### PR DESCRIPTION
**Pull request by xdoardo**
(still a draft, but CI tests need to be run)

Closes #5077.

### CODE REVIEW

File: tests/integration/cli/tests/config.rs [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
-    WasmerConfig::default().save(&config_path).unwrap();
+    // The config path and the config contents themselves are manually crafted so that we don't
+    // depend on the cli crate.
+    //
+    // Eventually, this part of the config shall live on a freestanding crate - perhaps added to
+    // `wasmer-config`.
+    let config_path = temp.path().join("wasmer.toml");
+
+    let contents = r#"
+telemetry_enabled = false
```
Comment (syrusakbary) [inline tests/integration/cli/tests/config.rs:19]: Why false by default. This should be true

File: tests/integration/cli/tests/config.rs [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
+update_notifications_enabled = false
```
Comment (syrusakbary) [inline tests/integration/cli/tests/config.rs:20]: This should be true as well

File: tests/integration/cli/tests/config.rs (hunk 1 above)
Comment (xdoardo) [reply tests/integration/cli/tests/config.rs:19]: It's the default we have when we create the config, both [in the `registry` crate](https://github.com/wasmerio/wasmer/blob/main/lib/registry/src/config.rs#L11) and [in the `cli` crate](https://github.com/wasmerio/wasmer/blob/main/lib/cli/src/config/mod.rs#L33)(which is what we currently use). Should I change the default to `true` there?

File: lib/cli/src/commands/connect.rs [hunk 3]
```diff
@@ -36,65 +36,65 @@ pub struct CmdConnect {
 impl AsyncCliCommand for CmdConnect {
     type Output = ();
 
-    #[cfg(all(target_os = "linux", feature = "tun-tap"))]
-    async fn run_async(mut self) -> Result<(), anyhow::Error> {
-        use edge_schema::{AppId, NetworkIdEncodingMethod, WELL_KNOWN_VPN};
-        use virtual_mio::Selector;
+    //#[cfg(all(target_os = "linux", feature = "tun-tap"))]
```
Comment (theduke) [inline lib/cli/src/commands/connect.rs:39]: Add a note why this is commented out.

File: lib/cli/src/commands/create_exe.rs [hunk 4]
```diff
@@ -192,9 +195,12 @@ pub struct Volume {
...
 }
 
-impl CreateExe {
+#[async_trait::async_trait]
+impl AsyncCliCommand for CreateExe {
+    type Output = ();
+
     /// Runs logic for the `compile` subcommand
-    pub fn execute(&self) -> Result<()> {
+    async fn run_async(self) -> Result<Self::Output, anyhow::Error> {
```
Comment (theduke) [inline lib/cli/src/commands/create_exe.rs:203]: I think this can just implement CliCommand, no need for async?

**Review by theduke — COMMENTED** [review]
> +1,134 /  −11,102 

That's what you like to see!

**Review by theduke — APPROVED** [review]

**Pull request merged**